src/foam/nanos/dig/drivers/DigFormatDriver.js

```diff
@@ -179,7 +179,11 @@ foam.CLASS({
       long pageSize = DigFormatDriver.MAX_PAGE_SIZE;
       if ( ! SafetyUtil.isEmpty(limit) ) {
         long l = Long.valueOf(limit);
-        if ( l != AbstractDAO.MAX_SAFE_INTEGER && l < pageSize) {
+        if ( l == 0 ) {
```

Comment:
Before, the MAX_PAGE_SIZE was 1000 and we would only let you decrease the limit, not increase it. But now you want to allow removing the limit entirely by setting it to zero. This could potentially cost excessive server load and create http requests too large to be served. I'm okay with increasing the MAX_PAGE_SIZE, but I don't think this change is a good idea without also adding permissioning. Also, we should test to see what size limits are achievable without causing issues using the various DIG supported formats.

Reply:
@kgrgreer .. I did some performance testing and found that we can retrieve ~400,000 JSON records and ~500,000 XML records.

```
+-------------------+---------------+-------------+---------------+------------+
| Count of Accounts | Time (JSON)   | Size (JSON) | Time (XML)    | Size (XML) |
+-------------------+---------------+-------------+---------------+------------+
| 200,000           | 26.97 s       | 204.22 MB   | 19.04 s       | 125.33 MB  |
+-------------------+---------------+-------------+---------------+------------+
| 300,000           | 38 s          | 301.95 MB   | 29 s          | 185.3 MB   |
+-------------------+---------------+-------------+---------------+------------+
| 400,000           | Out Of Memory | -           | 37 s          | 245 MB     |
+-------------------+---------------+-------------+---------------+------------+
| 500,000           | -             | -           | Out Of Memory | -          |
+-------------------+---------------+-------------+---------------+------------+
```